runtime/near-evm-runner/src/builtins.rs

```diff
@@ -0,0 +1,587 @@
+use std::{
+    cmp::{max, min},
+    io::{self, Cursor, Read},
+    mem::size_of,
+};
+
+use byteorder::{BigEndian, ByteOrder, LittleEndian, ReadBytesExt};
+use ethereum_types::{Address, H256, U256};
+use num_bigint::BigUint;
+use num_traits::{FromPrimitive, One, ToPrimitive, Zero};
+use parity_bytes::BytesRef;
+use ripemd160::Digest;
+use vm::{MessageCallResult, ReturnData};
+
+#[derive(Primitive)]
+enum Precompile {
+    EcRecover = 1,
+    Sha256 = 2,
+    Ripemd160 = 3,
+    Identity = 4,
+    ModexpImpl = 5,
+    Bn128AddImpl = 6,
+    Bn128MulImpl = 7,
+    Bn128PairingImpl = 8,
+    Blake2FImpl = 9,
+    LastPrecompile = 10,
+}
+
+pub fn is_precompile(addr: &Address) -> bool {
+    *addr < Address::from_low_u64_be(Precompile::LastPrecompile.to_u64().unwrap())
+}
+
+pub fn precompile(id: u64) -> Result<Box<dyn Impl>, String> {
+    Ok(match Precompile::from_u64(id) {
+        Some(Precompile::EcRecover) => Box::new(EcRecover) as Box<dyn Impl>,
+        Some(Precompile::Sha256) => Box::new(Sha256) as Box<dyn Impl>,
+        Some(Precompile::Ripemd160) => Box::new(Ripemd160) as Box<dyn Impl>,
+        Some(Precompile::Identity) => Box::new(Identity) as Box<dyn Impl>,
+        Some(Precompile::ModexpImpl) => Box::new(ModexpImpl) as Box<dyn Impl>,
+        Some(Precompile::Bn128AddImpl) => Box::new(Bn128AddImpl) as Box<dyn Impl>,
+        Some(Precompile::Bn128MulImpl) => Box::new(Bn128MulImpl) as Box<dyn Impl>,
+        Some(Precompile::Bn128PairingImpl) => Box::new(Bn128PairingImpl) as Box<dyn Impl>,
```

Comment:
why these are bn128, found current fork use bn256: https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go#L75

Reply:
Damn, indeed it's updated.
I've tried to update the version of EVM but the versions of all of the libraries didn't work... I guess need to do it to actually properly support.